proxy/http/remap/PluginFactory.cc

```diff
@@ -171,13 +171,15 @@ PluginFactory::getRemapPlugin(const fs::path &configPath, int argc, char **argv,
     plugin = new RemapPluginInfo(configPath, effectivePath, runtimePath);
     if (nullptr != plugin) {
       if (plugin->load(error)) {
-        PluginDso::loadedPlugins()->add(plugin);
-
         if (plugin->init(error)) {
+          PluginDso::loadedPlugins()->add(plugin);
           inst = RemapPluginInst::init(plugin, argc, argv, error);
```

Comment:
> Suppose the new instance fails - what stops this configuration load from proceeding

If `PluginFactory::getRemapPlugin()` call returned `nullptr`  at [RemapConfig.cc:812](https://github.com/apache/trafficserver/blob/master/proxy/http/remap/RemapConfig.cc#L812) it would fail the plugin loading and then it would abort the config load in `remap_parse_config_bti()` immediately.

`PluginFactory::getRemapPlugin()` initializes `inst` with `nullptr` at the beginning so if any of the following fails it would return `inst=nullptr`

- `plugin->init()` (which calls `TSRemapInit()`) 
- `plugin->load()` (which loads the DSO)

The failing of the config reload was already there and working before this PR, this PR fixes a discrepancy of not actually unloading the DSO in the case of a failing `plugin->init()` (`TSRemapInit()`)